### PR TITLE
feat: output quality based on browser viewport

### PR DESCRIPTION
**Pull request by doppiam**
added the capabilities to control the output quality based on the viewport of the browser, to do it, just launch the browser at a specific viewport
```typescript
 const browser = await launchBrowser({
    viewport: {
      width: 1280,
      height: 720,
    },
  });
```

### CODE REVIEW

**Comment by doppiam** [comment]
## Streaming size

The streaming output will be at the same size of the viewport (1920x1080 by default), so you can control the output size by specifying the browser's viewport size

```typescript
const browser = await launchBrowser({
  viewport: {
    width: 1280,
    height: 720,
  }
});
```

To start the streaming at a different size from the browser's viewport, specify width and height on `startStreaming`
```typescript
const {stop} = await startStreaming(page, {
  width: 1280,
  height: 720,
});

File: README.md [hunk 1]
```diff
@@ -44,6 +44,27 @@ await browser.close();
...
+});
+```
+
+To start the streaming at a different size from the browser's viewport, specify width and height on `startStreaming`
+```typescript
+const {stop} = await startStreaming(page, {
+  width: 1280,
+  height: 720,
+});
+```
```
Comment (Takeno) [inline README.md]: I'd like to add the way to change viewport from the default using `page.setViewport`

File: src/browser.ts [hunk 2]
```diff
@@ -13,6 +13,12 @@ type LaunchBrowserOptions = {
 
 type StreamID = string;
 
+type StartStreamingOptions = {
+  port?: number;
```
Comment (Takeno) [inline src/browser.ts]: if we move to options object, maybe we need to think about better names? `port` => `wsPort`, `width` => `viewportWidth`, or `viewport: {width, height}` ?

File: README.md (hunk 1 above)
Comment (doppiam) [reply README.md]: to be able to change it after initialization?

Comment (Takeno) [reply README.md]: To be able to change viewport for different pages for multiple streams

File: src/browser.ts [hunk 3]
```diff
@@ -13,6 +13,11 @@ type LaunchBrowserOptions = {
 
 type StreamID = string;
 
+type StartStreamingOptions = {
+  wsPort?: number;
+  viewport: {width?: number; height?: number};
```
Comment (Takeno) [inline src/browser.ts]: Probably it is better to keep both number required. `viewport` property optional?

File: src/browser.ts [hunk 4]
```diff
@@ -44,16 +49,34 @@ export async function launchBrowser({
...
+export async function startStreaming(
+  page: Page,
+  {
+    wsPort = 8080,
+    viewport = {
+      width: undefined,
+      height: undefined,
+    },
+  }: StartStreamingOptions = {
+    viewport: {width: undefined, height: undefined},
```
Comment (Takeno) [inline src/browser.ts]: Change above, here you just need to extract `viewport` and check if it is undefined or not.

File: src/chrome-extension/options.ts [hunk 5]
```diff
@@ -8,10 +8,20 @@ type RecordingSetup = {
 const recordings = new Map<string, RecordingSetup>();
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-async function startStreaming(port: number) {
+async function startStreaming(port: number, width?: number, height?: number) {
```
Comment (Takeno) [inline src/chrome-extension/options.ts]: Is it accepting the same parameters described by `type StartStreamingOptions`?

**Review by Takeno — CHANGES_REQUESTED** [review]

File: src/chrome-extension/options.ts [hunk 6]
```diff
@@ -8,10 +8,22 @@ type RecordingSetup = {
 const recordings = new Map<string, RecordingSetup>();
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-async function startStreaming(port: number) {
+async function startStreaming(port: number, recordingResizeFactor: number = 1) {
```
Comment (Takeno) [inline src/chrome-extension/options.ts]: Is it accepting the same parameters described by type `StartStreamingOptions`?

Comment (doppiam) [reply src/chrome-extension/options.ts]: yes

Comment (Takeno) [reply src/chrome-extension/options.ts]: then, let's use the `StartStreamingOptions` type

File: README.md [hunk 7]
```diff
@@ -44,6 +44,27 @@ await browser.close();
...
+
+To start the streaming at a different size from the browser's viewport, specify the `recordingResizeFactor` on `startStreaming` (default to `1`)
+```typescript
+const {stop} = await startStreaming(page, {
+  recordingResizeFactor: 0.5
+});
+```
+Here an example of [two different streaming size](./examples/record-to-file/different-outputs.ts).
```
Comment (Takeno) [inline README.md]: This example doesn't exist anymore

**Review by Takeno — CHANGES_REQUESTED** [review]

**Review by Takeno — APPROVED** [review]

**Pull request merged**